src/webargs/multidictproxy.py

```diff
@@ -0,0 +1,67 @@
+from webargs.compat import MARSHMALLOW_VERSION_INFO, Mapping
+from webargs.core import missing, is_multiple
+
+
+class MultiDictProxy(Mapping):
+    """
+    A proxy object which wraps multidict types along with a matching schema
+    Whenever a value is looked up, it is checked against the schema to see if
+    there is a matching field where `is_multiple` is True. If there is, then
+    the data should be loaded as a list or tuple.
+
+    In all other cases, __getitem__ proxies directly to the input multidict.
+    """
+
+    def __init__(self, multidict, schema):
+        self.data = multidict
+        self.multiple_keys = self._list_multiple_keys(schema)
+
+    def _list_multiple_keys(self, schema):
+        result = []
+        for name, field in schema.fields.items():
+            if not is_multiple(field):
+                continue
+            if MARSHMALLOW_VERSION_INFO[0] < 3:
+                result.append(field.load_from or name)
+            else:
+                result.append(field.data_key or name)
```

Comment:
`data_key` can be an empty string

```suggestion
                result.append(field.data_key if field.data_key is not None else name)
```

Same with `load_from`.

Just fixed this in marshmallow (https://github.com/marshmallow-code/marshmallow/pull/1379), so it's fresh in mind 😄

Reply:
I see the truthiness of your words. (See what I did there? :wink: )
I think this is the only place now that inspects individual fields on the schemas, but I'll double check to make sure this isn't an issue elsewhere.